data-reconciliation-app/src/services/authentication-service.ts

```diff
@@ -67,215 +26,37 @@ export interface IAuthenticationService {
-        if (isValid.success && isValid.content) {
-          return ServiceResult.Succeeded(identityId);
-        }
-      }
-      // check if caller is a valid member
-      case CcfAuthenticationPolicyEnum.Member_cert:
-      case CcfAuthenticationPolicyEnum.Member_signature: {
-        const memberCaller = request.caller as unknown as UserMemberAuthnIdentity;
-        const identityId = memberCaller.id;
-        const isValid = this.isActiveMember(identityId);
-        if (isValid.success && isValid.content) {
-          return ServiceResult.Succeeded(identityId);
-        }
-      }
-    }
-
-    return ServiceResult.Failed({
-      errorMessage: "Error: invalid caller identity",
-      errorType: "AuthenticationError"
-    });
-  }
-
-  /**
-   * Checks if a user exists
-   * @see https://microsoft.github.io/CCF/main/audit/builtin_maps.html#users-info
-   * @param {string} userId userId to check if it exists
-   * @returns {ServiceResult<boolean>}
-   */
-  public isUser(userId: string): ServiceResult<boolean> {
     try {
-      if (!userId) {
-        return ServiceResult.Failed({
-          errorMessage: "Error: invalid user id",
-          errorType: "AuthenticationError",
-        });
-      }
-
-      const usersCerts = ccfapp.typedKv(
-        "public:ccf.gov.users.certs",
-        ccfapp.arrayBuffer,
-        ccfapp.arrayBuffer
-      );
-      const result = usersCerts.has(ccf.strToBuf(userId));
-      return ServiceResult.Succeeded(result);
-    } catch (ex) {
-      return ServiceResult.Failed({
-        errorMessage: "Error getting caller identity",
-        errorType: "AuthenticationError",
-        details: ex,
-      });
-    }
-  }
-
-  /**
-   * Checks if a member exists and active
-   * @see https://microsoft.github.io/CCF/main/audit/builtin_maps.html#members-info
-   * @param {string} memberId memberId to check if it exists and active
-   * @returns {ServiceResult<boolean>}
-   */
-  public isActiveMember(memberId: string): ServiceResult<boolean> {
-    try {
-      if (!memberId) {
-        return ServiceResult.Failed({
-          errorMessage: "Error: invalid member id",
-          errorType: "AuthenticationError",
-        });
-      }
-
-      const membersCerts = ccfapp.typedKv(
-        "public:ccf.gov.members.certs",
-        ccfapp.arrayBuffer,
-        ccfapp.arrayBuffer
-      );
-
-      const isMember = membersCerts.has(ccf.strToBuf(memberId));
-
-      const membersInfo = ccfapp.typedKv(
-        "public:ccf.gov.members.info",
-        ccfapp.arrayBuffer,
-        ccfapp.arrayBuffer
-      );
-
-      const memberInfoBuf = membersInfo.get(ccf.strToBuf(memberId));
-      const memberInfo = ccf.bufToJsonCompatible(memberInfoBuf) as CCFMember;
-      const isActiveMember = memberInfo && memberInfo.status === "Active";
+      const commonCaller = request.caller as unknown as ccfapp.AuthnIdentityCommon;
+      const authPolicy = this.validators.get(<CcfAuthenticationPolicyEnum>commonCaller.policy);
```

Comment:
You are actually getting a `validator` here and not `authPolicty`. So this can be better

```typescript
const validator = this.validators.get(<CcfAuthenticationPolicyEnum>commonCaller.policy);
return validator.validate(request);
```